### ISSUE TITLE
button:  well-done

### ISSUE THREAD

**Issue by paul-shannon** [mariam16548/learningShiny#1]
Hi @mariam16548,   Nice work on the button example.  Well done.

Before moving onto more javascript, it will be wise to learn some CSS.

With some judicious web searching, figure out how to

    - wrap your two html elements (div and button) with a surrounding div
    - figure out the css ("style") which will center your elements in the middle of the page, 
       both top to bottom, and right to left, such that when you resize the web browser,
       they stay in the center

**Comment by paul-shannon** [comment]
@mariam16548  please use this github issues mechanism for communication about code

**Comment by mariam16548** [comment]
Hi @paul-shannon, I will get to work on that!

**Comment by paul-shannon** [comment]
@mariam16548 wrote (off list):  
````
I've also uploaded the diagram and three paragraphs about UDP, TCP, and IP on to GitHub (learningShiny/notes/networkNotesDiagram.png). I will continue learning more about CSS.
````
Here are some suggestions:

1.  Your entire document is a png.  That makes little sense.  It is not editable, no text copy and paste is possible. It cannot be revised.    A google doc would be better, then shared with me, on which I can make comments.   You can paste in the graphics as graphics without losing the editable text of the document.  Please do that - then I will comment, make suggestions.
2.  Take care to arrange your html document - and all code - so that the structure of the document is reflected in the formatting of the text in the file.  Specifically:  tags should be indented 3 or 4 spaces for each  nested element.  Those at the same nested level will then then line up vertically in your text file, making it easy to read, easy to understand.   You did this well for "demo" div, but <button> is not aligned at all.   
3.  It is not fatal in a script this short, but it will soon become costly.   A div named "demo" and a function named "myFunction" tells the reader nothing about their purpose.  These names should be descriptive.  Good names play an indispensable role in documenting code.  
4.  Once the above 3 items are fixed, the next concept for you to grasp is the client/server relationship in HTTP -  in preparation for moving to shiny.    Study up on HTTP "GET" - that is a good entry point into the subject.    Discover the difference between peer/peer IPC relations (typically done with websockets), and client/server which centralized databases and webservers use.

**Comment by mariam16548** [comment]
@paul-shannon Thank you for the feedback! 

1. I have shared the diagram as a Google doc (you should have gotten an email, but I also have this [link](https://docs.google.com/document/d/1qFlEOp0TIGFdJotZ9zH559wpKpu0OOWwxAOcymhEYnA/edit?usp=sharing)).
2. I will work on making sure to name variables appropriately!
3. I will look into HTTP "GET" and explore related concepts!

**Comment by paul-shannon** [comment]
@mariam16548  wrote
````
 I've also uploaded the diagram and three paragraphs about UDP, TCP, and IP on to GitHub (learningShiny/notes/networkNotesDiagram.png). I will continue learning more about CSS. 
````

**Comment by paul-shannon** [comment]
Hi @mariam16548 

I added my own versions of two introductory paragraphs to your document.   
These are the expository principles I tried to follow
1.  For structure:  Short, simple sentences, containing just one principal idea
2.  Provide context.  Every idea only makes sense in context.  Find ways to compactly provide that for the expected range of readers.
Good writing is hard to do!   But good writing encourages good thinking.  "Here is what I want to say, here is what I am trying to understand, this is the context in which it is important."    Code is the same:  good organization of clearly identified functions, clarity for the reader.
